### ISSUE TITLE
Temporary Execution

### ISSUE THREAD

**Issue by CMCDragonkai** [fregster/PHPSandbox#2]
Hello,

Is there a way to modify the code so the php isn't executed from sample files? This is because the environment I'm in has an ephemeral filesystem, so I need to use code from the database or code that is executed temporarily as part of the file itself.

I saw that you use the command line shell to do a lot of the execution, will it work with just raw code?

**Comment by fregster** [comment]
Yes 'technically' running raw code is an option but it did not seem to work well in testing and if I remember it's not able to lint the code prior to exec'ing it which means you'll get the fatal error. You can obviously catch it and deal with it. The bigest problem was safely passing the code to the process, base64 would be okay but could potentially hit the max command length's on some OS's. If you could send me a sample code string I'd look at getting it working for you.

**Comment by CMCDragonkai** [comment]
Hey Fregster, in the time span between me asking you the question, and you answering has allowed me to actually take your PHPSandbox and completely change it to fit my requirements. (Well not completely change it).

I think you may find some of them as improvements as you can pipe in the code and I combined it with lint, whitelist, blacklist, stactic code analysis.. etc.

Probably won't work well as a pull request as the architecture is different. I'm using Codeigniter.

One thing though, I removed chroot from the options as I was scouring the internet for documentation on it, and it didn't seem to do anything.

**Comment by fregster** [comment]
I'd love to see the code, perhaps we could merge it to a better project.

Paul

**Comment by CMCDragonkai** [comment]
Sure I have to finish up the static code analysis right now. I've hit a problem trying to do an array key tree search (http://stackoverflow.com/questions/13692816/searching-for-an-array-key-branch-inside-a-larger-array-tree-php/)

**Comment by fregster** [comment]
Would be really easy in XML but with an array I see your problem.

I'll think about it.

Paul
